src/Core/Services/AuthService.cs

```diff
@@ -218,10 +218,18 @@ public async Task<AuthResult> LogInPasswordlessAsync(string email, string access
             }
             else
             {
-                var decKey = await _cryptoService.RsaDecryptAsync(masterKey, decryptionKey);
-                var decKeyHash = await _cryptoService.RsaDecryptAsync(masterKeyHash, decryptionKey);
-                response = await LogInHelperAsync(email, accessCode, Encoding.UTF8.GetString(decKeyHash), null, null, null, new MasterKey(decKey), null, null,
-                null, null, authRequestId: authRequestId);
+                if (string.IsNullOrEmpty(masterKeyHash) && decryptionKey != null)
+                {
+                    var decryptedKey = await _cryptoService.RsaDecryptAsync(masterKey, decryptionKey);
```

Comment:
⛏️ This line gets repeated in the `else` as well, so it can be moved before the `if` given that it's used on both flows.